### PR TITLE
Makes IUser extensible

### PR DESCRIPTION
**Pull request by paynecrl97**
#1025 

This PR adds `IEntity` support to `IUser` and moves user create/edit logic from the `AdminController` and into a driver. This allows the `IUser` object to be extended.

I'll raise a further PR at some point shortly to change the admin list view of users to use drivers as well.

### CODE REVIEW

**Comment by sebastienros** [comment]
Just a little request, could we add new permissions named `ManageUserInformation` and `ManageOwnUserInformation` (which is implied by the first one)? When editing a user we then check for `ManageUserInformation`. The `ManageOwnInformation` is provided by a permission handler (like for content item Ownership permissions). 

Also the driver should check for specific permission in Edit and Update methods, for instance Roles can't be edited but by someone with a specific Roles permission.

Maybe we should also remove the generic ManageUsers one and split it in Create/Edit/Delete exactly as we do for content items.

If you prefer we do it in a separate PR then that's fine.

**Comment by paynecrl97** [comment]
I'll do that in a separate PR so that it doesn't block these changes from being merged.

File: src/OrchardCore/OrchardCore.Users.Abstractions/IUser.cs [hunk 1]
```diff
@@ -1,9 +1,11 @@
+using OrchardCore.Entities;
+
 namespace OrchardCore.Users
 {
     /// <summary>
     /// Marker interface for ASP.NET Core Identity services.
     /// </summary>
-    public interface IUser
+    public interface IUser : IEntity
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Users.Abstractions/IUser.cs]: Not sure about it. It'd be better to keep it at the class level. Only the Users module needs to know about this IMO, even if it means casting sometimes.

Comment (paynecrl97) [reply src/OrchardCore/OrchardCore.Users.Abstractions/IUser.cs]: Ah I thought I had kept it to the Users module (the namespace on this class doesn't match the project it is in). I'll change it.

**Comment by Ermesx** [comment]
The great example of extendibility of an entity!

**Comment by sebastienros** [comment]
And now it can also support custom parts like the site settings do ...

**Pull request merged**